src/function/cast/string_cast.cpp

```diff
@@ -166,9 +167,73 @@ bool StringListCastLoop(string_t *source_data, ValidityMask &source_mask, Vector
 	       all_converted;
 }
 
-bool StringListCast(Vector &source, Vector &result, idx_t count, CastParameters &parameters) {
+static LogicalType InitVarcharStructType(const LogicalType &target) {
+	return LogicalType::STRUCT(
+	    child_list_t<LogicalType>(StructType::GetChildCount(target), make_pair("", LogicalType::VARCHAR)));
+}
+
+bool VectorStringToStruct::StringToNestedTypeCastLoop(string_t *source_data, ValidityMask &source_mask, Vector &result,
+                                                      ValidityMask &result_mask, idx_t count,
+                                                      CastParameters &parameters, const SelectionVector *sel) {
+
+	auto varchar_struct_type = InitVarcharStructType(result.GetType());
+	Vector varchar_vector(varchar_struct_type, count);
+	auto &child_vectors = StructVector::GetEntries(varchar_vector);
+	auto &result_children = StructVector::GetEntries(result);
+
+	string_map_t<idx_t> child_names;
+	vector<ValidityMask *> child_masks;
+	// to cover missing values, set all child masks to invalid, then only set the found values to valid
+	for (idx_t child_idx = 0; child_idx < result_children.size(); child_idx++) {
+		child_names.insert({StructType::GetChildName(result.GetType(), child_idx), child_idx});
+		if (result.GetVectorType() == VectorType::CONSTANT_VECTOR) {
```

Comment:
I could be misunderstanding, but isn't this either never the case or always?
I believe you always receive a result Vector as a certain type (I believe it's CONSTANT, but could be FLAT)

Could you check the code coverage results? If those hit both lines then you can completely discard anything I said :)

Reply:
Maybe we can take a look at the code coverage together tomorrow? I haven't quite got the hang of it yet. I explicitly added this bit of code because I ran into an assertion. So my assumption was that the children have same the vectortype as their parent

Reply:
I thought this was the entry point to your cast, and you were not in control of the result vector being passed, apologies.
I think you're right in that struct children have the same vectortype as the struct vector itself